### PR TITLE
examples: add Distroless

### PR DESCRIPTION
**Pull request by heasterday**
Addresses #1139

### CODE REVIEW

**Comment by heasterday** [comment]
Blocked by #1170

File: examples/distroless/Dockerfile
Comment (j-ogas) [inline examples/distroless/Dockerfile:4]: Of the distroless tags what reason, if any, do we pick the python3 one? Perhaps comment on our choice here.

**Review by j-ogas — CHANGES_REQUESTED** [review]

File: examples/distroless/Dockerfile
Comment (heasterday) [reply examples/distroless/Dockerfile:4]: I chose the`python3` image because of all of their base images it is the simplest to test. I will add a comment to this effect.

File: examples/distroless/Dockerfile
Comment (reidpr) [inline examples/distroless/Dockerfile:9]: Interestingly once PR #698 is merged, this will become no longer needed.

File: examples/distroless/test.bats
Comment (reidpr) [inline examples/distroless/test.bats:4]: I think these four lines are not needed? They aren't used and the test title prints out properly.

File: examples/distroless/Dockerfile
Comment (reidpr) [inline examples/distroless/Dockerfile:6]: I suggest (a) copy only `hello.py` since that's all that's used, and (2) put it at the root directory.

**Review by reidpr — CHANGES_REQUESTED** [review]
Nice work, a few minor suggestions.

File: examples/distroless/Dockerfile
Comment (heasterday) [reply examples/distroless/Dockerfile:9]: Would you like me to block this PR pending #698 ?

File: examples/distroless/test.bats
Comment (heasterday) [reply examples/distroless/test.bats:4]: Every other `test.bats` I looked at has these lines, is this something that should by tidied up outside of this specific test?

File: examples/distroless/Dockerfile
Comment (heasterday) [reply examples/distroless/Dockerfile:6]: This was done to be consistent with the `hello` and `mpihello` tests. Is this something that you would like corrected in the other tests?

File: examples/distroless/Dockerfile
Comment (reidpr) [reply examples/distroless/Dockerfile:9]: No, that one will take a while.

File: examples/distroless/test.bats
Comment (reidpr) [reply examples/distroless/test.bats:4]: Probably. Some might actually need it, or maybe it's completely obsolete.

File: examples/distroless/Dockerfile
Comment (reidpr) [reply examples/distroless/Dockerfile:6]: Some of the others do put multiple files in. I would leave the others alone for now.

File: examples/distroless/test.bats
Comment (heasterday) [reply examples/distroless/test.bats:4]: I'll create a separate issue to track this.

Comment (heasterday) [reply examples/distroless/test.bats:4]: It looks like the `true` may not be needed. Without line 3 there is this error:  
```
 ✗ NO_TAG_SET/hello
   (in test file /home/heasterday/charliecloud/examples/distroless/test.bats, line 13)
     `[[ $status -eq 0 ]]' failed
   ch-run[657842]: can't find image: /var/tmp/images/NO_TAG_SET: No such file or directory (ch-run.c:146 2)
   ```  
And line 2 suppresses this shellcheck warning
```
   CH_TEST_TAG=$ch_test_tag
^---------^ SC2034: CH_TEST_TAG appears unused. Verify use (or export if used externally).
```

Comment (reidpr) [reply examples/distroless/test.bats]: Bizarre; it must be some `ch-test` magic I've forgotten about.

The `true` is needed: without it, the `shellcheck disable` applies to the whole file, not just the subsequent line.

Maybe these gotchas should be documented in the contributors' guide?

Comment (heasterday) [reply examples/distroless/test.bats:4]: Added back the `true`. Documenting the process for adding test does indeed sound like a good idea.

**Review by reidpr — APPROVED** [review]

**Pull request merged**
